DuckDuckGo/BrowserChromeManager.swift

```diff
@@ -147,6 +144,10 @@ class BrowserChromeManager: NSObject, UIScrollViewDelegate {
 
 private class BarsAnimator {
     
+    struct Constants {
+        static let bottomThreshold: CGFloat = 110
```

Comment:
I wonder, where does this number come from?

Reply:
In "third scenario" there is a comment:

> In Safari, bars will not disappear when user scrolls downwards just above bottom of the web page (threshold estimated to 110, see Constants.bottomThreshold).

I just compared same page on Safari and DDG side by side, this value looks very close.

Reply:
Wouldn't that number be equal to the combined height of the areas covered by top & bottom bars (since hiding bars within that distance of the bottom would have to readjust content offset to accommodate for expanded viewport)? 

Do you think that makes sense? If so, we could probably get rid of the constant and use `delegate` to obtain `omniBar.bounds.height` and `toolbarHeight` to use instead.

Reply:
I see the idea. But won't we also need bottom safe area which is part of bottom bar but is not part of `toolbarHeight`?

The value will be 52 + 44 + 34.

BTW, In this interaction Safari freezes bottom bar in transitioning state when "bottomThreshold" entered and animates hide on touch up. I like current DDG implementation better 😄

Reply:
Yes, if device has safe area then this value will be different - what I meant was that if possible, I'd prefer to use actual values coming from the properties of a view, rather than constants. 

And yes, implementation seems to work rather smoothly. :)

Reply:
Updated it. Actually it worked well without additional safe area inset.